### PR TITLE
feat: add enableDAB flag to enable and disable DAB features

### LINKED ISSUE #15231: Add enableDAB flag to enable and disable DAB features
Set the flag to false for release/0.53

### PR DESCRIPTION
**Pull request by iwsimon**
**Description**:

Set the flag to false for release/0.53
Added HapiTests to check the flag.

**Related issue(s)**:

Fixes #15231

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeCreateHandler.java [hunk 1]
```diff
@@ -130,6 +129,7 @@ public void handle(@NonNull final HandleContext handleContext) {
     @NonNull
     @Override
     public Fees calculateFees(@NonNull final FeeContext feeContext) {
+        checkDABEnable(feeContext);
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeCreateHandler.java:132]:
```suggestion
        checkDABEnabled(feeContext);
```

File: hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/FreezeHandler.java [hunk 2]
```diff
@@ -167,7 +165,7 @@ public void handle(@NonNull final HandleContext context) throws HandleException
         final var filesConfig = context.configuration().getConfigData(FilesConfig.class);
 
         final FreezeUpgradeActions upgradeActions = new FreezeUpgradeActions(
-                adminServiceConfig, freezeStore, freezeExecutor, upgradeFileStore, nodeStore, stakingInfoStore);
+                context, freezeStore, freezeExecutor, upgradeFileStore, nodeStore, stakingInfoStore);
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/FreezeHandler.java:168]: Seems like context is not needed

File: hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/FreezeUpgradeActions.java [hunk 3]
```diff
@@ -39,21 +39,21 @@ public class FreezeUpgradeActions extends ReadableFreezeUpgradeActions {
...
+     * @param context the HandleContext
      * @param freezeStore the freeze store
      * @param executor the executor
      * @param upgradeFileStore the upgrade file store
      * @param nodeStore the node store
      * @param stakingInfoStore the staking info store
      */
     public FreezeUpgradeActions(
-            @NonNull final NetworkAdminConfig adminServiceConfig,
+            @NonNull final HandleContext context,
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/FreezeUpgradeActions.java:50]: Just Configuration is fine to be passed in

File: hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/ReadableFreezeUpgradeActions.java [hunk 4]
```diff
@@ -85,20 +89,44 @@ public class ReadableFreezeUpgradeActions {
     public static final String MARK = "✓";
 
     public ReadableFreezeUpgradeActions(
-            @NonNull final NetworkAdminConfig adminServiceConfig,
+            @NonNull final ConfigProvider configProvider,
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/ReadableFreezeUpgradeActions.java:92]: We can change everything to pass just `Configuration` , so we don't need new constructor with `HandleContext`

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeCreateHandler.java (hunk 1 above)
Comment (iwsimon) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeCreateHandler.java]: Updated.

File: hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/FreezeHandler.java (hunk 2 above)
Comment (iwsimon) [reply hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/FreezeHandler.java]: Yes, changed to use Configuration, good suggest. Thanks @Neeharika-Sompalli

File: hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/FreezeUpgradeActions.java (hunk 3 above)
Comment (iwsimon) [reply hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/FreezeUpgradeActions.java]: Updated.

File: hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/ReadableFreezeUpgradeActions.java (hunk 4 above)
Comment (iwsimon) [reply hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/ReadableFreezeUpgradeActions.java]: Changed.

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM! tyvm @iwsimon

**Review by netopyr — APPROVED** [review]
LGTM - tyvm @iwsimon

**Review by tinker-michaelj — APPROVED** [review]
All makes sense to me, tyvm @iwsimon !

**Pull request merged**